kong/router/init.lua

```diff
@@ -10,6 +10,117 @@ local atc_compat = require("kong.router.atc_compat")
 local utils = require("kong.router.utils")
 local is_http = ngx.config.subsystem == "http"
 
+local DEFAULT_MINIMAL_INTERVAL = 60 -- 1 minute
+
+local phonehome_statistics
+local phonehome_sync_start
+do
+  local reports = require("kong.reports")
+  local semaphore = require("ngx.semaphore")
+  local nkeys = require("table.nkeys")
+  local empty_table = {}
+  -- reuse tables to avoid cost of creating tables and garbage collection
+  local protocol_counts = {
+    http = 0, -- { "http", "https" },
+    stream = 0, -- { "tcp", "tls", "udp" },
+    tls_passthrough = 0, -- { "tls_passthrough" },
+    grpc = 0, -- { "grpc", "grpcs" },
+  }
+  local path_handling_counts = {
+    v0 = 0,
+    v1 = 0,
+  }
+  local route_report = {
+    flavor = "unknown",
+    paths_count = 0,
+    headers_count = 0,
+    total = 0,
+    regex_paths_count = 0,
+    protocols = protocol_counts,
+    path_handling = path_handling_counts,
+  }
+
+
+  local need_report = assert(semaphore.new())
+
+  local function report_timer_impl(premature)
+    if premature then
+      return
+    end
+
+    if need_report:wait(DEFAULT_MINIMAL_INTERVAL) then
+      reports.send("routes", route_report)
+    end
+  end
+
+  local function traditional_statistics(routes)
+    for _, route in ipairs(routes) do
+      route = route.route
+      local paths = route.paths or empty_table
+      local headers = route.headers or empty_table
+      route_report.paths_count = route_report.paths_count + #paths
+      route_report.headers_count = route_report.headers_count + nkeys(headers)
```

Comment:
These variables should be local variables until the end where the `route_report` can be created/assigned. Please avoid table access in a loop.